apps/backoffice-v2/src/pages/Entity/components/Case/hooks/usePendingRevisionEvents/usePendingRevisionEvents.tsx

```diff
@@ -0,0 +1,60 @@
+import { TWorkflowById } from '@/domains/workflows/fetchers';
+import { useCallback, useMemo } from 'react';
+import { calculateAllWorkflowEvents } from '@/pages/Entity/components/Case/hooks/usePendingRevisionEvents/utils/calculate-pending-workflow-events';
+import { CommonWorkflowEvent } from '@ballerine/common';
+import { checkIsKybExampleVariant } from '@/lib/blocks/variants/variant-checkers';
+import { useRevisionCaseMutation } from '@/domains/workflows/hooks/mutations/useRevisionCaseMutation/useRevisionCaseMutation';
+import { IPendingEvent } from './interfaces';
+
+function composeUniqueWorkflowEvents(
+  acc: Record<string, IPendingEvent>,
+  pendingWorkflowEvent: IPendingEvent,
+) {
+  acc[`${pendingWorkflowEvent?.workflowId}-${pendingWorkflowEvent?.pendingEvent}`] =
+    pendingWorkflowEvent;
+
+  return acc;
+}
+
+function isRevisionEvent(pendingWorkflowEvent: IPendingEvent) {
+  return (
+    pendingWorkflowEvent?.pendingEvent === CommonWorkflowEvent.REVISION ||
+    pendingWorkflowEvent?.pendingEvent === CommonWorkflowEvent.TASK_REVIEWED
+  );
+}
+
+export const usePendingRevisionEvents = (
+  mutateRevisionCase: ReturnType<typeof useRevisionCaseMutation>['mutate'],
+  workflow?: TWorkflowById,
+) => {
+  const pendingWorkflowEvents = useMemo(() => {
+    if (!workflow) return;
+
+    return calculateAllWorkflowEvents(workflow);
+  }, [workflow]);
+
+  const onMutateRevisionCase = useCallback(() => {
+    if (!pendingWorkflowEvents || !workflow) return;
+
+    const uniqueWorkflowEvents = pendingWorkflowEvents
+      .filter((pendingWorkflowEvent: IPendingEvent) => isRevisionEvent(pendingWorkflowEvent))
+      .reduce((acc: Record<string, IPendingEvent>, pendingWorkflowEvent: IPendingEvent) => {
+        return composeUniqueWorkflowEvents(acc, pendingWorkflowEvent);
+      }, {});
```

Comment:
The logic here is very short and simple. Readability can remain the same if the logic is inline and not abstracted as long as curly braces are used with explicit returns.

Reply:
I removed the curlies